storage-operator/pkg/apis/storage/v1alpha1/volume_types.go

```diff
@@ -63,44 +63,72 @@ type PersistentVolumeTemplateSpec struct {
 
 type VolumePhase string
 
-// "Enum" representing the phase/status of a volume.
+// TODO: kept for temporary compatibility, to be removed.
+// "Enum" representing the phase of a volume.
 const (
 	VolumeFailed      VolumePhase = "Failed"
 	VolumePending     VolumePhase = "Pending"
 	VolumeAvailable   VolumePhase = "Available"
 	VolumeTerminating VolumePhase = "Terminating"
 )
 
-type VolumeErrorCode string
+type ConditionReason string
 
+// TODO: replace those by more fine-grained ones.
 // "Enum" representing the error codes of the Failed state.
 const (
-	InternalError    VolumeErrorCode = "InternalError"
-	CreationError    VolumeErrorCode = "CreationError"
-	DestructionError VolumeErrorCode = "DestructionError"
-	UnavailableError VolumeErrorCode = "UnavailableError"
+	ReasonPending     ConditionReason = "Pending"
+	ReasonTerminating ConditionReason = "Terminating"
+
+	ReasonInternalError    ConditionReason = "InternalError"
+	ReasonCreationError    ConditionReason = "CreationError"
+	ReasonDestructionError ConditionReason = "DestructionError"
+	ReasonUnavailableError ConditionReason = "UnavailableError"
+)
+
+type VolumeConditionType string
+
+const (
+	// VolumeReady means Volume is ready to be used.
+	VolumeReady VolumeConditionType = "Ready"
 )
 
+type ConditionStatus string
```

Comment:
Can't we reuse the type from `k8s.io/pkg/apis/core` (or whatever the right import path is)?

Reply:
Ho yes, you're right, let's use `corev1.ConditionStatus`